File: src/com/mrpowergamerbr/loritta/commands/vanilla/fun/NyanCatCommand.java
package com.mrpowergamerbr.loritta.commands.vanilla.fun;

import java.awt.Image;
import java.awt.image.BufferedImage;
import java.io.ByteArrayInputStream;
import java.io.ByteArrayOutputStream;
import java.io.File;
import java.io.InputStream;
import java.util.Arrays;
import java.util.List;

import javax.imageio.ImageIO;

import org.apache.commons.lang3.StringUtils;

import com.mrpowergamerbr.loritta.Loritta;
import com.mrpowergamerbr.loritta.commands.CommandBase;
import com.mrpowergamerbr.loritta.commands.CommandContext;

import net.dv8tion.jda.core.MessageBuilder;

public class NyanCatCommand extends CommandBase {
	@Override
	public String getLabel() {
		return "nyan";
	}

	@Override
	public String getDescription() {
		return "Nyan Cat, diretamente no seu servidor! E você pode pedir o tamanho do Nyan Cat igual quando você pede algum sanduíche no Subway!";
	}
	
	@Override
	public String getUsage() {
		return "cat";
	}
	
	@Override
	public List<String> getExample() {
		return Arrays.asList("", "cat", "caaaaaaat", "caaaaaaaaaaaaat");
	}
	
	@Override
	public void run(CommandContext context) {
		int times = 0;
		if (context.getArgs().length == 1) {
			times = StringUtils.countMatches(context.getArgs()[0], "a");
		}
		
		try {
			Image catLeft = ImageIO.read(new File(Loritta.FOLDER + "cat_left.png"));
			Image catRight = ImageIO.read(new File(Loritta.FOLDER + "cat_right.png"));
			Image catMiddle = ImageIO.read(new File(Loritta.FOLDER + "cat_middle.png"));
			Image catMiddle2 = ImageIO.read(new File(Loritta.FOLDER + "cat_middle2.png"));
			Image catMiddle3 = ImageIO.read(new File(Loritta.FOLDER + "cat_middle3.png"));
			Image catMiddle4 = ImageIO.read(new File(Loritta.FOLDER + "cat_middle4.png"));
			Image catMiddle5 = ImageIO.read(new File(Loritta.FOLDER + "cat_middle5.png"));

			BufferedImage bi = new BufferedImage(catLeft.getWidth(null) + catRight.getWidth(null) + (catMiddle.getWidth(null) * times), catLeft.getHeight(null), BufferedImage.TYPE_INT_ARGB);

			int x = 0;

			bi.getGraphics().drawImage(catLeft, x, 0, null);

			x += catLeft.getWidth(null);

			int idx = 0;

			while (times > idx) {
				int c = Loritta.getRandom().nextInt(0, 5);
				if (c == 0) {
					bi.getGraphics().drawImage(catMiddle, x, 0, null);
				}
				if (c == 1) {
					bi.getGraphics().drawImage(catMiddle2, x, 0, null);
				}
				if (c == 2) {
					bi.getGraphics().drawImage(catMiddle3, x, 0, null);
				}
				if (c == 3) {
					bi.getGraphics().drawImage(catMiddle4, x, 0, null);
				}
				if (c == 4) {
					bi.getGraphics().drawImage(catMiddle5, x, 0, null);
				}
				x += catMiddle.getWidth(null);
				idx++;
			}

			bi.getGraphics().drawImage(catRight, x, 0, null);

			ByteArrayOutputStream os = new ByteArrayOutputStream();
			ImageIO.write(bi, "png", os);
			InputStream is = new ByteArrayInputStream(os.toByteArray());

			MessageBuilder builder = new MessageBuilder();
			builder.append(context.getAsMention(true));
			context.sendFile(is, "nyan_cat.png", builder.build());
		} catch (Exception e) {
			e.printStackTrace();
		}
	}

}

File: src/main/java/com/mrpowergamerbr/loritta/commands/vanilla/fun/NyanCatCommand.kt
package com.mrpowergamerbr.loritta.commands.vanilla.`fun`

import com.mrpowergamerbr.loritta.Loritta
import com.mrpowergamerbr.loritta.commands.CommandBase
import com.mrpowergamerbr.loritta.commands.CommandContext
import com.mrpowergamerbr.loritta.utils.ImageUtils
import com.mrpowergamerbr.loritta.utils.f
import com.mrpowergamerbr.loritta.utils.locale.BaseLocale
import org.apache.commons.lang3.StringUtils
import java.awt.Color
import java.awt.image.BufferedImage
import java.io.File
import javax.imageio.ImageIO

class NyanCatCommand : CommandBase() {
	override fun getLabel(): String {
		return "nyan"
	}

	override fun getDescription(locale: BaseLocale): String {
		return locale.NYANCAT_DESCRIPTION.f()
	}

	override fun getUsage(): String {
		return "cat"
	}

	override fun getExample(): List<String> {
		return listOf("", "cat", "caaaaaaat", "caaaaaaaaaaaaat")
	}

	override fun needsToUploadFiles(): Boolean {
		return true
	}

	override fun run(context: CommandContext) {
		var times = 0
		if (context.args.size == 1) {
			times = StringUtils.countMatches(context.args[0], "a")
		}

		val catLeft = ImageIO.read(File(Loritta.FOLDER + "cat_left_v2.png"))
		val catRight = ImageIO.read(File(Loritta.FOLDER + "cat_right_v2.png"))
		val catMiddle = ImageIO.read(File(Loritta.FOLDER + "cat_middle_v2.png"))

		val bi = BufferedImage(catLeft.getWidth(null) + catRight.getWidth(null) + catMiddle.getWidth(null) * times, catLeft.getHeight(null), BufferedImage.TYPE_INT_ARGB)

		var x = 0

		bi.graphics.drawImage(catLeft, x, 0, null)

		x += catLeft.getWidth(null)

		var idx = 0

		while (times > idx) {
			val catMiddleCopy = BufferedImage(catMiddle.width, catMiddle.height, BufferedImage.TYPE_INT_ARGB);
			val graphics = catMiddleCopy.graphics;
			graphics.drawImage(catMiddle, 0, 0, null); // Nós iremos "clonar" o nosso cat middle para colocar alguns pontinhos rosas aleatórios :)
			graphics.color = Color(255, 51, 153); // Usar a cor rosa-meio-roxo que o Nyan Cat tem

			val randomDots = Loritta.random.nextInt(0, 6);

			val invalidDotsPositionsX = ArrayList<Int>();
			val invalidDotsPositionsY = ArrayList<Int>();

			for (i in 0..randomDots) {
				val randomX = Loritta.random.nextInt(0, 2);
				val randomY = Loritta.random.nextInt(2, 16);
				if (invalidDotsPositionsX.contains(randomX) || invalidDotsPositionsX.contains(randomX - 1) || invalidDotsPositionsX.contains(randomX + 1)) {
					continue;
				}

				if (invalidDotsPositionsY.contains(randomY) || invalidDotsPositionsY.contains(randomY - 1) || invalidDotsPositionsY.contains(randomY + 1)) {
					continue;
				}

				// Sabia que width/height 0 = 1px? Agora você sabe!
				graphics.drawRect(randomX, randomY, 0, 0); // Faça um retângulo 1x1 na nossa coordenada aleatória

				invalidDotsPositionsX.add(randomX);
				invalidDotsPositionsY.add(randomY);
			}

			invalidDotsPositionsX.clear();
			invalidDotsPositionsY.clear();

			bi.graphics.drawImage(catMiddleCopy, x, 0, null) // E depois desenhe nossa imagem modificada na imagem original
			x += catMiddleCopy.getWidth(null)
			idx++
		}

		bi.graphics.drawImage(catRight, x, 0, null)

		context.sendFile(ImageUtils.toBufferedImage(bi.getScaledInstance(bi.width * 4, bi.height * 4, BufferedImage.SCALE_AREA_AVERAGING)), "nyan_cat.png", context.getAsMention(true))
	}
}